packages/apps/job-launcher/server/src/modules/cron-job/cron-job.repository.ts

```diff
@@ -1,34 +1,44 @@
-import { Injectable, Logger } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
-import { FindOneOptions, FindOptionsWhere, Repository } from 'typeorm';
+import { Injectable } from '@nestjs/common';
+import { CronJobType } from 'src/common/enums/cron-job';
+import { DataSource, QueryFailedError, Repository } from 'typeorm';
+import { handleQueryFailedError } from '../../database/database.error';
 import { CronJobEntity } from './cron-job.entity';
-import { CronJobType } from '../../common/enums/cron-job';
 
 @Injectable()
-export class CronJobRepository {
-  private readonly logger = new Logger(CronJobRepository.name);
+export class CronJobRepository extends Repository<CronJobEntity> {
+  constructor(private dataSource: DataSource) {
+    super(CronJobEntity, dataSource.createEntityManager());
+  }
 
-  constructor(
-    @InjectRepository(CronJobEntity)
-    private readonly cronJobEntityRepository: Repository<CronJobEntity>,
-  ) {}
+  async createUnique(type: CronJobType): Promise<CronJobEntity> {
```

Comment:
Why don't we just use base repo here?

Reply:
For that repository, I made the change in its own PR https://github.com/humanprotocol/human-protocol/pull/1552

Reply:
Ok.